app/controllers/insights_cloud/settings_controller.rb

```diff
@@ -9,14 +9,6 @@ def update
       render_setting(:insightsSyncEnabled, :allow_auto_insights_sync)
     end
 
-    def save_token_and_sync
```

Comment:
in case we drop this line and there's no API token provided, the user won't be able to access the insights page
as they can't change the token... I would suggest reverting back to keep this OR deleting also the Insights `NoTokenEmptyState.js` file and usages.